types/json/binary_functions.go

```diff
@@ -51,6 +51,8 @@ func (bj BinaryJSON) Type() string {
 		return "DOUBLE"
 	case TypeCodeString:
 		return "STRING"
+	case TypeCodeOpaque:
+		return "OPAQUE"
```

Comment:
not all opaque type will return "OPAQUE" directly, it sometimes returns `BLOB` or `BIT`. e.g.
```sql
mysql> select json_type(cast(0x41 as json));
+-------------------------------+
| json_type(cast(0x41 as json)) |
+-------------------------------+
| BLOB                          |
+-------------------------------+
```

Reply:
Fixed.

`select json_type(cast(0x41 as json));` runs on MySQL 5.7.39 could return `BIT`, but in your comment (and TiDB situation) it will return `BLOB`. But I think it has nothing to do with the `JsonOpaque`.